### PR TITLE
Chore: implementing start animation feauture

### PR DESCRIPTION
**Pull request by szymonkadas**
# Summary:
implemented start animation. With little tweaks in the future it should work backwards aswell. 

# Notes: 
There's a little problem with text inside TitleBadge twitching when its parent container (circle) is increasing its size. 
To be honest, I have no idea for now how to fix it.

### CODE REVIEW

File: src/components/SplashScreen.tsx [hunk 1]
```diff
@@ -7,21 +8,41 @@ type SplashScreenProps = {
...
 export default function SplashScreen(props: SplashScreenProps) {
+  const [isVisible, setIsVisible] = useState(true);
+  const [transition, setTransition] = useState(false);
+  // w przyszłości można to przerobić tak by tranzycja odpalała się na starcie tak jak prawdopodobnie wygladałaby odwrócona animacja po złożeniu zamówienia. możnaby wtedy zrobić logikę na zasadzie przełącznika.
+  const handleTransition = () => {
+    // start transition
+    setTransition(true);
+    // mogą być potrzebne refy do tego timeoutu.
+    setTimeout(() => {
+      setIsVisible(false);
```
Comment (Sumick) [inline src/components/SplashScreen.tsx]: **nitpick:** This logic is missing screen switching. It's only hiding splash screen. I would strongly suggest to implement some router here, to make it easy to navigate through your app. You can use `react-router` for example

File: src/components/SplashScreen.tsx [hunk 2]
```diff
@@ -7,21 +8,41 @@ type SplashScreenProps = {
...
+      setTransition(false);
+    }, 4000);
+    // for test purposes
+    setTimeout(() => {
+      setIsVisible(true);
```
Comment (Sumick) [inline src/components/SplashScreen.tsx]: **nitpick:** If it's for testing, we should remove it before merging

File: src/components/splashScreen/Circle.tsx [hunk 3]
```diff
@@ -2,9 +2,16 @@ import styles from "./Circle.module.css";
...
   columnLayout: boolean;
+  transition: boolean;
   children?: React.ReactNode;
 };
 
 export default function Circle(props: CircleProps) {
-  return <div className={props.columnLayout ? styles.columnCircle : styles.rowCircle}>{props.children}</div>;
+  let circleLayout = styles.columnCircle;
+  let circleEscapeLayout = styles.columnCircleEscape;
+  if (props.columnLayout === false) {
```
Comment (Sumick) [inline src/components/splashScreen/Circle.tsx]: **suggestion:** You can simplify it to:

```tsx
export default function Circle({ columnLayout = true, transition, children }: CircleProps) {
  const circleLayout = columnLayout ? styles.columnCircle : styles.rowCircle;
  const circleEscapeLayout = transition ? (columnLayout ? styles.columnCircleEscape : styles.rowCircleEscape) : '';

  return <div className={`${circleLayout} ${circleEscapeLayout}`}>{children}</div>;
}
```

File: src/components/splashScreen/TitleBadge.tsx [hunk 4]
```diff
@@ -1,14 +1,36 @@
...
   isPaniniOrdered: boolean;
+  handleTransition: () => void;
 };
 
 export default function TitleBadge(props: TitleBadgeProps) {
+  const [isButtonAvailable, setIsButtonAvailable] = useState(true);
+  const handleClick = () => {
+    setIsButtonAvailable(false);
+    setTimeout(() => {
+      setIsButtonAvailable(true);
```
Comment (Sumick) [inline src/components/splashScreen/TitleBadge.tsx]: **thought:** I don't think this logic is necessary, button should only run handleTransition handler in my opinion

File: src/components/splashScreen/TitleBadge.tsx [hunk 5]
```diff
@@ -1,14 +1,36 @@
...
+    <div className={`${styles.titleBadge} ${!isButtonAvailable && styles.titleBadgeEscape}`}>
       <h1 className={styles.textContent}>Panini {props.isPaniniOrdered ? "ordered" : "Creator"}</h1>
-      <button className={styles.button}>{props.isPaniniOrdered ? "start again" : "begin"}</button>
+      <button
+        className={styles.button}
+        onClick={() => {
+          props.handleTransition();
+          handleClick();
+        }}
+        disabled={isButtonAvailable ? false : true}
```
Comment (Sumick) [inline src/components/splashScreen/TitleBadge.tsx]: **nitpick:** You can simplify it to: 

```ts
  disabled={isButtonAvailable}
```

File: src/components/splashScreen/TitleBadge.tsx [hunk 6]
```diff
@@ -1,14 +1,36 @@
...
+    cursor: "initial",
+  };
   return (
-    <div className={styles.titleBadge}>
+    <div className={`${styles.titleBadge} ${!isButtonAvailable && styles.titleBadgeEscape}`}>
       <h1 className={styles.textContent}>Panini {props.isPaniniOrdered ? "ordered" : "Creator"}</h1>
-      <button className={styles.button}>{props.isPaniniOrdered ? "start again" : "begin"}</button>
+      <button
+        className={styles.button}
+        onClick={() => {
```
Comment (Sumick) [inline src/components/splashScreen/TitleBadge.tsx]: **nitpick:** Writing inline handlers is considered a bad practice - you can put this `props.handleTransition` into `handleClick`, so you can do:

```tsx
 onClick={handleClick}
```

File: src/components/splashScreen/TitleBadge.tsx [hunk 7]
```diff
@@ -1,14 +1,36 @@
...
 
 export default function TitleBadge(props: TitleBadgeProps) {
+  const [isButtonAvailable, setIsButtonAvailable] = useState(true);
+  const handleClick = () => {
+    setIsButtonAvailable(false);
+    setTimeout(() => {
+      setIsButtonAvailable(true);
+    }, 4000);
+  };
+  const buttonStyle = {
```
Comment (Sumick) [inline src/components/splashScreen/TitleBadge.tsx]: **nitpick:** It's considered a bad practice to put styles object inside component body, as this object reference has to be recreated on every render, it leads to worse app performance (of course when it's overused). You can fix it by putting this object outside component body, so it's created only once

**Review by Sumick — CHANGES_REQUESTED** [review]
We need few improvements + there is one requirement missing - we should implement screen transition in this task scope. It means that when SplashScreen is gone, we should display FormScreen (with some initial setup, because form UI will be created later). I suggest you to implement routing mechanism - it will be very proper solution then. Check react router as a very popular example of such solution: https://reactrouter.com/en/main/start/tutorial

File: src/App.tsx [hunk 8]
```diff
@@ -1,14 +1,21 @@
...
-  return (
-    <div>
-      <SplashScreen isPaniniOrdered={Object.keys(orderData).length > 0 ? true : false}></SplashScreen>
-    </div>
+  const router = createBrowserRouter(
+    createRoutesFromElements(
+      <Route path="/" element={<Layout isPaniniOrdered={Object.keys(orderData).length > 0 ? true : false} />}>
+        <Route index element={<SplashScreen shouldTransition={false} />} />
+        <Route path="/panini_creator" element={<PaniniCreatorLayout shouldTransition={true} />}></Route>
+        <Route path="/panini_creator:options" element={<PaniniCreatorLayout shouldTransition={false} />}></Route>
```
Comment (Sumick) [inline src/App.tsx]: **question:** Why do you need this route?

**Review by Sumick — CHANGES_REQUESTED** [review]
In my last review I wrote you `we should implement screen transition in this task scope. It means that when SplashScreen is gone, we should display FormScreen (with some initial setup, because form UI will be created later)`. You've created UI components which is not in this task scope, because it is specified for later. In this task scope you just needed routing mechanism, and screens change - nothing else. Please, leave at this PR only task related changes. You can cherry pick your commit with these components changes, and use it later in next tasks. Pull requests should not be that big.

File: src/App.tsx (hunk 8 above)
Comment (szymonkadas) [reply src/App.tsx]: # Explanation: 
About the options route, I've figured out that the state of form (order) would be preserved in the url (so the user could share it with friends, and come back to his configuration with just url, so it's kinda future functionality. 
Also for now, the animation turns on automatically without any params, so i've figured out such workaround,
# Thoughts: 
 Maybe better approach would be to navigate user (from splashscreen) to path with :transitioned_param for transition animation? And then create subRoute for order (with inputs customised by user, through url)?

**Comment by szymonkadas** [comment]
Reverted surplus changes. Code should now contain only necessary changes in this feature scope.

**Review by Sumick — APPROVED** [review]
Looks good!

**Pull request merged**